packages/shared/src/lib/kratos.ts

```diff
@@ -246,12 +246,10 @@ export const getKratosError = async (id: string): Promise<ErrorData> => {
 export const checkKratosEmail = async (
   email: string,
 ): Promise<KratosEmailData> => {
-  const res = await fetch(
-    `${heimdallUrl}/api/check_email?email_address=${email}`,
-    {
-      method: 'POST',
-    },
-  );
+  const params = new URLSearchParams({ email });
+  const res = await fetch(`${heimdallUrl}/api/check_email?${params}`, {
+    method: 'POST',
+  });
```

Comment:
Why not use the whole `URL` interface? Like this if `params` was empty you would also get trailing `?`.

Reply:
I think params can't be empty here, at most what can happen is `email` is empty, which would result in `/check_email?email=`

I'm open to updating this, though the end result would be the same?

Reply:
Actually worse, if `email` was undefined the result would be `/check_email?email=undefined`. But either way, don't see why not to use dedicated URL API to assemble the URL.

Reply:
oh wow, you're right, damn JS man 😆 how is that a thing?

Reply:
Because URLSearchParams (and URL as well) calls `toString` when adding params through constructor or `params.append`. And `toString` of `undefined` is `"undefined"` 😆